pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs

```diff
@@ -171,32 +184,67 @@ private void PopulateComboBoxes()
+            SetColumnPositions(Settings.Default.CustomImportTransitionListColumnsList);
+
+            // Make a copy of the current transition list with 100 rows or the length of the current transition list (whichever is smaller)
+            int length = Math.Min(Importer.RowReader.Lines.Count, 100);
+            IList<string> lines = new List<string>();
+            for (int i = 0; i < length; i++)
             {
-                SetComboBoxText(columns.DecoyColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Decoy);
-                SetComboBoxText(columns.IrtColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_iRT);
-                SetComboBoxText(columns.LabelTypeColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Label_Type);
-                SetComboBoxText(columns.LibraryColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Library_Intensity);
-                SetComboBoxText(columns.PeptideColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Peptide_Modified_Sequence);
-                SetComboBoxText(columns.PrecursorColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Precursor_m_z);
-                SetComboBoxText(columns.ProductColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Product_m_z);
-                SetComboBoxText(columns.ProteinColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Protein_Name);
-                SetComboBoxText(columns.FragmentNameColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Fragment_Name);
-                // Commented out for consistency because there is no product charge column
-                // SetComboBoxText(columns.PrecursorChargeColumn, Resources.ImportTransitionListColumnSelectDlg_PopulateComboBoxes_Precursor_Charge);  
+                lines.Add(Importer.RowReader.Lines[i]);
+            }
+            // Try importing that list to check for errors
+            MassListInputs input = new MassListInputs(lines);
+            var insertionParams = new DocumentChecked();
+            List<TransitionImportErrorInfo> testErrorList1 = null;
+            insertionParams.Document = _docCurrent.ImportMassList(input, Importer, null,
+                _insertPath, out insertionParams.SelectPath, out insertionParams.IrtPeptides,
+                out insertionParams.LibrarySpectra, out testErrorList1, out insertionParams.PeptideGroups);
+
+            var allError = ReferenceEquals(insertionParams.Document, _docCurrent);
+            // If all transitions are errors, reset the columns to the detected columns
+            if (allError)
+            {
+                SetColumnPositions(detectedColumns);
             }
         }
+        // Returns the current column positions as a list of tuples
+        private List<Tuple<int, string>> CurrentColumnPositions()
```

Comment:
Think I prefer a struct with field names to a Tuple<> here, especially with the comment below explaining Item1 and Item2. Better to have self-documenting code when possible.